src/ipc/handler.c

```diff
@@ -118,6 +118,9 @@ struct sof_ipc_cmd_hdr *mailbox_validate(void)
 {
 	struct sof_ipc_cmd_hdr *hdr = ipc_get()->comp_data;
 
+	/* reset and clear the stale IPC data */
+	bzero(hdr, SOF_IPC_MSG_MAX_SIZE);
```

Comment:
Why does it need cleared, wont the new data overwtire the old data ?

Reply:
the new data will overwrite the size of the new IPC only, if the old data size is larger than the old one, the addition stale part could be read as the new IPC data by mistake when the ABI b/w FW and driver is not exactly matched.

Reply:
But hdr.size menas this wont happen. i.e. we always read hdr.size (and ignore any extra data)

Reply:
the hdr.size is used to copy from mailbox to the buffer, so there is no stale mailbox data copied to the buffer. but the stale extra data in the buffer itself could be used inside the FW by mistake, e.g. when we are reading extended data from the offset, we might use the size of the extended data struct which could be inconsistent with hdr.size (hdr.size != offset + sizeof(extended data)), reading the extended data there could get some staled data from the buffer.

Reply:
hdr.size is always the size of the **whole** message including any optional extended data. 

If FW or the driver is doing something different for hdr.size in some cases then its a bug. What happens if this PR is not applied ?

Reply:
The PR is help for backward/forward compatibility only, with it applied, we can handle the FW ABI is newer than the driver one issue (e.g. FW is in 3.20 whild driver is in 3.19), without it, in this mis-matched scenario, FW will either report an mis-matched/hdr.size error, or reading a staled data.

Reply:
The fact the we need this means your kernel PR is not backawards compatible i.e. it's a MAJOR ABI change.  Please tell me which of the following statements your PR breaks.

1) FW will only ever read hdr.size of data. 
2) FW will check for offset > hdr.size from ABI 3.17 onwards. 
3) UUID is mandatory in topology for ABI >= 3.17
4) Kernel with ABI >= 3.17 AND topology with ABI>=3.17 will append exteneded data (and increase hdr.size).
5) Kernel ABI >= 3.17 with topology < 3.17 does not send extended data.
6) Kernel ABI < 3.17 with topology >= 3.17 does not send extended data.

Reply:
@lgirdwood as discussed in the meeting, this is not mandatory if the driver for the new added extended data handler can validate this extended data size mismatch(for different ABI on FW and driver) correctly, let me close the PR.